spring-cloud-aws-sns/src/test/java/io/awspring/cloud/sns/sms/core/SnsSmsTemplateIntegrationTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.sms.core;
+
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.SNS;
+
+import io.awspring.cloud.sns.sms.attributes.SmsMessageAttributes;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sns.SnsClient;
+
+/**
+ * @author Matej Nedic
+ * @since 3.0.0
+ */
+@Testcontainers
+class SnsSmsTemplateIntegrationTest {
+	private static SnsSmsTemplate snsSmsTemplate;
+	private static SnsClient snsClient;
+
+	@Container
+	static LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:0.14.0")).withServices(SNS).withReuse(true);
+
+	@BeforeAll
+	public static void createSnsTemplate() {
+		snsClient = SnsClient.builder().endpointOverride(localstack.getEndpointOverride(SNS))
+				.region(Region.of(localstack.getRegion()))
+				.credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("noop", "noop")))
+				.build();
+		snsSmsTemplate = new SnsSmsTemplate(snsClient);
+	}
+
+	@Test
+	void sendValidMessage_ToPhoneNumber() {
+		Assertions
+				.assertDoesNotThrow(() -> snsSmsTemplate.send("+385 00 000 0000", "Spring Cloud AWS got you covered!"));
```

Comment:
Can we assert log message from Localstack?

https://github.com/localstack/localstack/pull/2822

Reply:
At first glance, I am not seeing it in container logs but will check again later.

Reply:
If I understand correctly, we're only checking the message is sent and doesn't throw any errors. Maybe we can fetch the message from local stack after sending it, and assert that?

If there are any concurrency concerns, I usually use a `CountDownLatch` to wait on the test thread until the message is consumed.

Reply:
Hey @tomazfernandes I tried with getting localstack logs but I am not seeing a message. We can on other hand make operations and template return messageId and check that. Localstack uses moto to mock sms sending.

Reply:
Hey @MatejNedic, nice work here! Sure, if the SNS API returns an id I guess we should return that to users.

This being an SNS integration, can't we maybe subscribe to the topic and receive the sent message ourselves too? Just guessing, have no clue how this actually works 😄

Reply:
@MatejNedic Did you set the DEBUG env var to "1"? in LocalStackContainer should be `.withEnv("DEBUG", "1")` and change the logger for testcontainers to DEBUG too `<logger name="org.testcontainers" level="DEBUG"/>` just to make sure something useful is in the container's log. Then you can fetch the logs doing `String logs = localstack.getLogs(OutputFrame.OutputType.STDOUT, OutputFrame.OutputType.STDERR);` and assert against it.

Reply:
@eddumelendez I tried that as well but I am not seeing the correct message I want to see.
`Delivering SMS message to`. I am seeing `Publishing message to TopicArn: None | Message: Spring Cloud AWS got you covered` which makes sense but is still not what I want. I have added tests with at least some kind of message. I tried this against real AWS and I am getting SMS on my phone number without an issue.

@tomazfernandes we have something similar for SNS notifications where we send it to an SQS and we poll it from SQS. Sadly with SMS we can't do that.